skbio/stats/distance/tests/test_base.py

```diff
@@ -57,6 +64,112 @@ def setUp(self):
+                            ['d', 'e', 7.]],
+                           columns=['i', 'j', 'value'])
+
+        obs = self.dm_5x5.between(['b', 'd'], ['a', 'c', 'e'])
+        pdt.assert_frame_equal(obs, exp)
+
+    def test_between_order_stability(self):
+        exp = pd.DataFrame([['b', 'a', 5.],
+                            ['b', 'c', 6.],
+                            ['b', 'e', 8.],
+                            ['d', 'a', 4.],
+                            ['d', 'c', 6.],
+                            ['d', 'e', 7.]],
+                           columns=['i', 'j', 'value'])
+
+        # varying the order of the "i" values, result remains consistent
+        # with the test_between result
+        obs = self.dm_5x5.between(['d', 'b'], ['a', 'c', 'e'])
+        pdt.assert_frame_equal(obs, exp)
+
+        # varying the order of the "j" values, result remains consistent
+        # with the test_between result
+        obs = self.dm_5x5.between(['b', 'd'], ['a', 'e', 'c'])
+        pdt.assert_frame_equal(obs, exp)
+
+    def test_between_overlap(self):
+        exp = pd.DataFrame([['b', 'a', 5.],
+                            ['b', 'd', 7.],
+                            ['b', 'e', 8.],
+                            ['d', 'a', 4.],
+                            ['d', 'd', 0.],
+                            ['d', 'e', 7.]],
+                           columns=['i', 'j', 'value'])
+
+        # 'd' in i and j overlap
+        with self.assertRaisesRegex(KeyError, ("This constraint can "
+                                               "removed with "
+                                               "allow_overlap=True.")):
+            self.dm_5x5.between(['b', 'd'], ['a', 'd', 'e'])
+
+        obs = self.dm_5x5.between(['b', 'd'], ['a', 'd', 'e'],
+                                  allow_overlap=True)
+        pdt.assert_frame_equal(obs, exp)
+
+    def test_between_missing_id(self):
+        with self.assertRaisesRegex(MissingIDError, "not found."):
+            self.dm_3x3.between(['x', 'a'], ['a', 'b', 'c'])
+
+        with self.assertRaisesRegex(MissingIDError, "not found."):
+            self.dm_3x3.between(['a', 'b'], ['a', 'x', 'c'])
+
+        with self.assertRaisesRegex(MissingIDError, "not found."):
+            self.dm_3x3.between(['a', 'y'], ['a', 'x', 'c'])
+
+    def test_stable_order(self):
+        exp = np.array([1, 3, 4], dtype=int)
+        obs = self.dm_5x5._stable_order(['d', 'e', 'b'])
+
+        npt.assert_equal(obs, exp)
+
+    def test_subset_to_dataframe(self):
+        exp = pd.DataFrame([['b', 'a', 5.],
+                            ['b', 'd', 7.],
+                            ['b', 'e', 8.],
+                            ['d', 'a', 4.],
+                            ['d', 'd', 0.],
+                            ['d', 'e', 7.]],
+                           columns=['i', 'j', 'value'])
+
+        obs = self.dm_5x5._subset_to_dataframe([1, 3], [0, 3, 4])
+        pdt.assert_frame_equal(obs, exp)
```

Comment:
Is passing in an empty np.array for i and/or j to `_subset_to_dataframe()` a thing that could conceivably happen? If so, would be worth adding tests for that case (I imagine this would just result in an empty output).

If this isn't a thing that should ever happen, then it would be worth adding a quick check to the start of `_subset_to_dataframe()` that checks to make sure both `i_indices` and `j_indices` have at least one element (and raises an error if not).

Reply:
Good catch on the edge case, it does get slightly weird